src/writer/direntPool.h

```diff
@@ -43,9 +43,28 @@ namespace zim
         DirentPool(const DirentPool&) = delete;
         DirentPool& operator=(const DirentPool&) = delete;
         ~DirentPool() {
-          for(auto direntArray: pools) {
-            delete[] direntArray;
+          auto nbPools = pools.size();
+          if (nbPools == 0) {
+            return;
           }
+          // Delete all but last pools (add call the destructors of the dirents)
+          for (auto i = 0U; i<nbPools-1; i++) {
+            auto pool = pools[i];
+            for (auto j = 0U; j < 0xFFFF; j++) {
+              try {
+                pool[j].~Dirent();
+              } catch (...){ /*discard */ }
+            }
+            delete [] (reinterpret_cast<char*>(pool));
+          }
+          // On the last pool, only `direntIndex` are really constructed.
+          auto lastPool = pools[nbPools-1];
+          for (auto j = 0U; j<direntIndex; j++) {
+            try {
+              lastPool[j].~Dirent();
+            } catch (...){ /* discard */ }
+          }
+          delete [] (reinterpret_cast<char*>(lastPool));
```

Comment:
This can be extracted into a helper function `static void destroyPoolBlock(Dirent* pool, uint16_t count)` and reused.

Reply:
Done